storage/sql.go

```diff
@@ -96,11 +96,6 @@ func ReadTree(row Row) (*trillian.Tree, error) {
 	} else {
 		return nil, fmt.Errorf("unknown TreeType: %v", treeType)
 	}
-	if hs, ok := trillian.HashStrategy_value[hashStrategy]; ok {
-		tree.HashStrategy = trillian.HashStrategy(hs)
-	} else {
-		return nil, fmt.Errorf("unknown HashStrategy: %v", hashStrategy)
-	}
```

Comment:
After deleting this code, `hashStrategy` seems not used. Would here be a good place to verify that `hashStrategy == "RFC6962_SHA256"` for `[PREORDERED_]LOG` trees? As a safe-guard for the (unlikely) case that a log with a different hasher exists somewhere.

Reply:
`hashStrategy` is still somewhat necessary for the SQL query (I didn't want to fiddle with those, seemed easier and safer to still read/write them the same way), but good call on having a check to see that it matches what we expect it to be.

I'm not sure I get why the "for `[PREORDERED_]LOG` trees" bit? The only "known" `HashStrategy` is `RFC6962_SHA256`, so just generally check for that?

Reply:
Yeah, can make the check unconditional, since all trees are expected to be logs.

Reply:
Done, PTAL